samples/embedded-auth-with-sdk/src/main/java/com/okta/spring/example/controllers/LoginController.java

```diff
@@ -134,8 +145,67 @@ public ModelAndView selectAuthenticator(final @RequestParam("authenticator-type"
                                             final @RequestParam(value = "action") String action,
                                             final HttpSession session) {
 
+        logger.info("Auth Type: {}, Action: {}", authenticatorType, action);
+
         ProceedContext proceedContext = Util.getProceedContextFromSession(session);
 
+        if (authenticatorType != null && authenticatorType.equals("Security Key or Biometric")) {
+            IDXResponse idxResponse;
+            try {
+                List<Authenticator> authenticators = (List<Authenticator>) session.getAttribute("authenticators");
+
+                String authId = null;
+                for (Authenticator authenticator : authenticators) {
+                    if (authenticator.getLabel().equals("Security Key or Biometric")) {
+                        authId = authenticator.getId();
+                    }
+                }
+
+                AuthenticationResponse enrollResponse = idxAuthenticationWrapper.enrollAuthenticator(proceedContext, authId);
+                idxResponse = idxAuthenticationWrapper.getClient().
```

Comment:
In the past, I've added fields to `ProceedContext` when parsing the result, then adding another method to `AuthenticationWrapper` to pull the value out of `ProceedContext` or just adding a new field to the `AuthenticationResponse` object rather than having to call introspect again, and digging here manually.

Neither way is ideal, and this shows a limitation of the approach we've taken.

Reply:
💯